src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java

```diff
@@ -1,258 +1,512 @@
+        SecurityRule.Protocol udpProtocol = plugin.defineRuleProtocol(udpSecurityRule);
+        SecurityRule.Protocol icmpProtocol = plugin.defineRuleProtocol(icmpSecurityRule);
+        SecurityRule.Protocol anyProtocol = plugin.defineRuleProtocol(anySecurityRule);
+
+        // verify
+        Assert.assertEquals(SecurityRule.Protocol.TCP, tcpProtocol);
+        Assert.assertEquals(SecurityRule.Protocol.UDP, udpProtocol);
+        Assert.assertEquals(SecurityRule.Protocol.ICMP, icmpProtocol);
+        Assert.assertEquals(SecurityRule.Protocol.ANY, anyProtocol);
+    }
 
-        StatusLine statusLine = new BasicStatusLine(new ProtocolVersion("", 0, 0), httpStatus, "");
-        Mockito.when(httpResponse.getStatusLine()).thenReturn(statusLine);
+    // test case: when given an unsupported protocol, it should throw an exception
+    @Test(expected = FogbowException.class)
+    public void testDefineRuleProtocolUnsuccessful() throws FogbowException {
+        // setup
+        String nonExistingProtocol = "its-very-unlikelly-that-this-protocol-is-gonna-exist";
+        GetSecurityRulesResponse.SecurityRules tcpSecurityRule = createMockedSecurityRulesWithProtocol(nonExistingProtocol);
 
-        return httpResponse;
+        // verify
+        plugin.defineRuleProtocol(tcpSecurityRule);
+        Assert.fail();
     }
 
-    private NetworkOrder createNetworkOrder() throws Exception {
-        SystemUser systemUser = new SystemUser(FAKE_USER_ID, FAKE_USER_NAME, FAKE_TOKEN_PROVIDER);
+    private GetSecurityGroupsResponse createMockedSecurityGroupList(List<String> groupIds) {
+        GetSecurityGroupsResponse response = Mockito.mock(GetSecurityGroupsResponse.class);
+
+        List<GetSecurityGroupsResponse.SecurityGroup> securityGroups = new ArrayList();
+        for (String groupId : groupIds) {
+            securityGroups.add(createMockedSecurityGroup(groupId));
+        }
+        Mockito.when(response.getSecurityGroups()).thenReturn(securityGroups);
+        return response;
+    }
 
-        NetworkOrder order = new NetworkOrder(systemUser, FAKE_MEMBER_ID, FAKE_MEMBER_ID, FAKE_CLOUD_NAME,
-                FAKE_NAME, FAKE_GATEWAY, FAKE_ADDRESS, NetworkAllocationMode.STATIC);
+    private GetSecurityGroupsResponse.SecurityGroup createMockedSecurityGroup(String id) {
+        GetSecurityGroupsResponse.SecurityGroup securityGroupMock =
+                Mockito.mock(GetSecurityGroupsResponse.SecurityGroup.class);
+        Mockito.when(securityGroupMock.getId()).thenReturn(id);
+        return securityGroupMock;
+    }
 
-        NetworkInstance networtkInstanceExcepted = new NetworkInstance(order.getId());
-        order.setInstanceId(networtkInstanceExcepted.getId());
-        return order;
+    private SecurityRule createEmptySecurityRule() {
+        return new SecurityRule(SecurityRule.Direction.OUT, 0, 0, "0.0.0.0/0 ", SecurityRule.EtherType.IPv4, SecurityRule.Protocol.TCP);
```

Comment:
Use constants to define the CIDR and ports from the rule.

Reply:
Changed.